### PR TITLE
Raise exception if response from call() is CallError

### PR DESCRIPTION
**Pull request by tmh-azinhal**
This PR references issue #104 .

If the response is a CallError, an exception will be raised instead of being silenced.
To maintain backwards compatibility a `suppress=True` keyword argument was added in `Chargepoint.call`.

### CODE REVIEW

File: ocpp/charge_point.py [hunk 1]
```diff
@@ -224,7 +224,7 @@ async def _handle_call(self, msg):
             # when no '_on_after' hook is installed.
             pass
 
-    async def call(self, payload):
+    async def call(self, payload, surpress=True):
```
Comment (OrangeTux) [inline ocpp/charge_point.py:227]: s/surpress/suppress

Comment (OrangeTux) [reply ocpp/charge_point.py:227]: Can you also update the docstring and mention this new argument?

**Review by OrangeTux — CHANGES_REQUESTED** [review]
You can add a test? I'll come back later to

File: ocpp/charge_point.py (hunk 1 above)
Comment (tropxy) [reply ocpp/charge_point.py:227]: I would change the name to "suppress_ocpp_exception" or "suppress_exception". 

Or later on in the code I dont remember what the suppress is suppressing

**Review by tropxy — CHANGES_REQUESTED** [review]

File: ocpp/charge_point.py (hunk 1 above)
Comment (tmh-azinhal) [reply ocpp/charge_point.py:227]: I would say that keeping the name simple would be preferable. I hope the docstring explains what the argument's purpose is.

Comment (tropxy) [reply ocpp/charge_point.py]: ok, fair enough

**Comment by OrangeTux** [comment]
I promised some advice when it comes to writing tests for the piece of code you introduced.

In [tests/v16/test_v16_charge_point.py] you can add the tests. For the test you can reuse the `base_central_system` as is done in other tests in that file.

This test require a call, that is the request, and a call error. The latter is the response. Because of limitations of testing you first need prepare the response.  Create an instance of `ocpp.messages.CallError`, turn it into json and pass it to `base_central_system.route_message()`. [`route_message()`](https://github.com/mobilityhouse/ocpp/blob/master/ocpp/charge_point.py#L128) will put the response on [an internal response queue](https://github.com/mobilityhouse/ocpp/blob/master/ocpp/charge_point.py#L148).

Now you can create the request. You need to instantiate a payload from `call.v16.call` and 'send' that to the other end of the connection using `base_central_system.call()`. [`call()`](https://github.com/mobilityhouse/ocpp/blob/master/ocpp/charge_point.py#L227) will [use the response queue](https://github.com/mobilityhouse/ocpp/blob/master/ocpp/charge_point.py#L259-L260) to look for response to its request.

Note that it's important that the request and response have [the same unique id](https://github.com/mobilityhouse/ocpp/blob/master/ocpp/charge_point.py#L290-L291). Otherwise a response can't be matched to a request.

I hope this is enough information. If not, let me know.

**Review by OrangeTux — APPROVED** [review]

**Review by OrangeTux — CHANGES_REQUESTED** [review]
The lint job on CircleCI compains. Can you make the job happy?

**Review by OrangeTux — APPROVED** [review]
Thanks!

**Review by tropxy — APPROVED** [review]

**Pull request merged**